eurec4a_environment/nomenclature.py

```diff
@@ -0,0 +1,118 @@
+# (Vertical motion in the atmosphere has the standard name
+# upward_air_velocity.)
+ZONAL_WIND = "u"
+
+# From CF-conventions: "Northward" indicates a vector component which is
+# positive when directed northward (negative southward). Wind is defined as a
+# two-dimensional (horizontal) air velocity vector, with no vertical component.
+# (Vertical motion in the atmosphere has the standard name
+# upward_air_velocity.)
+MERIDIONAL_WIND = "v"
+
+
+CF_STANDARD_NAMES = {
+    TEMPERATURE: "air_temperature",
+    ALTITUDE: "geopotential_height",
+    RELATIVE_HUMIDITY: "relative_humidity",
+    POTENTIAL_TEMPERATURE: "potential_temperature",
+    PRESSURE: "air_pressure",
+    SPECIFIC_HUMIDITY: "specific_humidity",
+    WIND_SPEED: "wind_speed",
+    ZONAL_WIND: "eastward_wind",
+    MERIDIONAL_WIND: "northward_wind",
+}
+
+
+def _get_calling_function_name():
+    curframe = inspect.currentframe()
+    calframe = inspect.getouterframes(curframe, 2)
+    return calframe[1][3]
+
+
+class FieldMissingException(Exception):
+    pass
+
+
+def get_field(ds, field_name):
+    if field_name in ds:
+        return ds[field_name]
+
+    calling_function_name = _get_calling_function_name()
+
+    if field_name not in CF_STANDARD_NAMES:
+        raise FieldMissingException(
+            f"Couldn't find the variable `{field_name}` in the provided dataset."
+            f" To use {calling_function_name} you need to provide a variable"
+            f" with the name `{field_name}`"
+        )
+    else:
+        standard_name = CF_STANDARD_NAMES[field_name]
+        matching_dataarrays = {}
+        vars_and_coords = list(ds.data_vars) + list(ds.coords)
+        for v in vars_and_coords:
+            print(v)
+            if ds[v].attrs.get("standard_name", None) == standard_name:
+                matching_dataarrays[v] = ds[v]
+
+        if len(matching_dataarrays) == 0:
+            raise FieldMissingException(
+                f"Couldn't find the variable `{field_name}` in the provided dataset."
+                f" To use {calling_function_name} you need to provide a variable"
+                f" with the name `{field_name}` or set the `standard_name`"
+                f" attribute to `{standard_name}` for one or more of the variables"
+                " in the dataset"
+            )
+        elif len(matching_dataarrays) == 1:
+            return list(matching_dataarrays.values())[0]
+        else:
+            dims = [da.dims for da in matching_dataarrays.values()]
+            if len(set(dims)) == 1:
+                # all variables have the same dims, so we can create a
+                # composite data-array out of these
+                var_names = list(matching_dataarrays.keys())
+                var_dataarrays = list(matching_dataarrays.values())
+
+                # TODO: should we check units here too?
+                da_combined = xr.concat(var_dataarrays, dim="var_name")
```

Comment:
This is well thought of. Good implementation! However, all functions would have to be written to deal with this, i.e. multiple variables. e.g. estimating theta would require an array each of pressure and temperature. If `get_field` returns two temperature arrays and one pressure array, the function should know what to do... Ideally, return two theta arrays.  Did I understand that correctly?

Reply:
Thanks :smile: Yes, the idea would be that what is returned is always temperature for example, but it might be multiple temperature profiles. I've elaborated a bit on this idea in my comment below. Hope that makes sense...